src/main/java/seedu/taskify/logic/commands/EditCommand.java

```diff
@@ -85,10 +87,15 @@ public CommandResult execute(Model model) throws CommandException {
             throw new CommandException(MESSAGE_DUPLICATE_TASK);
         }
 
-        if (taskToEdit.getStatus().equals(new Status(StatusType.EXPIRED))) {
+        if (taskToEdit.getStatus().status.equals(StatusType.EXPIRED)
+                && !editedTask.getStatus().status.equals(StatusType.EXPIRED)) {
```

Comment:
Violation of Law of Demeter. Just use `getStatus().equals(new Status(StatusType.EXPIRED))`

Just realised, that the `status` field in `Status` should be private, not public.

Reply:
This one my bad, can change to:
private final StatusType statusType

Then add a getter method
public StatusType getStatusType()

I would change this myself but scared merge conflict harder for you to deal with later. Pm me if anything